src/map/mob.cpp

```diff
@@ -1413,31 +1413,66 @@ static int32 mob_ai_sub_hard_lootsearch(struct block_list *bl,va_list ap)
 }
 
 static int32 mob_warpchase_sub(struct block_list *bl,va_list ap) {
-	struct block_list *target;
-	struct npc_data **target_nd;
-	struct npc_data *nd;
+	block_list *target;
+	block_list **target_warp;
 	int32 *min_distance;
 	int32 cur_distance;
 
 	target= va_arg(ap, struct block_list*);
-	target_nd= va_arg(ap, struct npc_data**);
+	target_warp= va_arg(ap, struct block_list**);
 	min_distance= va_arg(ap, int32*);
 
-	nd = (TBL_NPC*) bl;
+	// NPC Warp
+	if (bl->type == BL_NPC) {
```

Comment:
Maybe consider using switch instead of if?

Reply:
Done.